### ISSUE TITLE
Node binary from Snap: Failed to load the ESLint library

### ISSUE THREAD

**Issue by kleinfreund** [microsoft/vscode-eslint#677]
- Node:
  - `which node`: /snap/bin/node
  - `node -v`: v12.2.0
- NPM:
  - `which npm`: /home/phil/.npm-global/bin/npm
  - `npm -v`: 6.9.0
  - `npm config get prefix`: /home/phil/.npm-global
- ESLint:
  - `which eslint`: /home/phil/.npm-global/bin/eslint
  - `eslint -v`: v5.16.0
- VSCode:
  - `which code`: /snap/bin/code
  - `code -v`: 1.34.0, a622c65b2c713c890fcf4fbf07cf34049d5fe758, x64

---

I checked the other issues that come up for this error, but from what I gather there, my setup should work.

For a newly created project that has pretty much only a JavaScript file (no package.json, no local node_modules), I get the “Failed to load the ESLint library” error. I have a second machine where for the same project, the ESLint library loads fine. I currently use a global installation of ESLint.

```
ESLint server stopped.
ESLint server running in node v10.2.0
ESLint server is running.
Failed to load the ESLint library for the document …
```

One difference between these two machines is that the former uses a Node binary distributed via Snap and the other uses one from Node’s PPA.

Now, I’m not quite sure if that’s a problem with `vscode-eslint` directly because I had other issues with the Node snap binary. Let me know if this should be raised elsewhere.

**Comment by dbaeumer** [comment]
@kleinfreund what is the output of `npm config get prefix`? And does the same happen when you specify `"eslint.runtime"="node"` as a setting?

**Comment by kleinfreund** [comment]
Running `npm config get prefix` yields `/home/phil/.npm-global`.

Specifying `"eslint.runtime": "node"` in VS Code and restarting changes the error:

```
[Info  - 2:34:34 PM] ESLint server stopped.
[Info  - 2:34:35 PM] Connection to server got closed. Server will restart.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:35 PM] Connection to server got closed. Server will restart.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:35 PM] Connection to server got closed. Server will restart.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:35 PM] ESLint server stopped.
[Info  - 2:34:36 PM] Connection to server got closed. Server will restart.
[Info  - 2:34:36 PM] ESLint server stopped.
[Info  - 2:34:36 PM] ESLint server stopped.
[Error - 2:34:36 PM] Connection to server got closed. Server will not be restarted.
[Info  - 2:34:36 PM] ESLint server stopped.
```

Additionally, the extension reports this in a notification:

```
The ESLint server crashed 5 times in the last 3 minutes. The server will not be restarted.
```

**Comment by kleinfreund** [comment]
Ah, I should’ve added the initial output from ESLint. It reports that the node runtime is node v10.2.0.

```
ESLint server stopped.
ESLint server running in node v10.2.0
ESLint server is running.
Failed to load the ESLint library for the document /home/phil/dev/repos/color-picker/js/color-picker.js
```

**Comment by dbaeumer** [comment]
That output is strange since it still uses the Electron's node version. Do you see the same when using `"eslint.runtime": "/snap/bin/node"`?

**Comment by kleinfreund** [comment]
Yes, I do. It still reports crashing ESLint server.

**Comment by dbaeumer** [comment]
Which version number do you see in:

```
ESLint server running in node ...
```

Do things work when node is not installed from snap?

**Comment by kleinfreund** [comment]
I only get `ESLint server running in node v10.2.0` when I don’t explicitly set `eslint.runtime` to either `node` or `/snap/bin/node`. Then, the node version is v10.2.0 as stated above. If I set the runtime, however, the output does not contain a line with a version number. Just the lines that I posted in https://github.com/microsoft/vscode-eslint/issues/677#issuecomment-493968620.

**Comment by dbaeumer** [comment]
OK, that means that very likely node doesn't launch at all. I am a little puzzled why when installing snaps this doesn't work. Does anyone have an idea what makes snaps different.

**Comment by kleinfreund** [comment]
@dbaeumer Does this issue require more information from my part?

**Comment by dbaeumer** [comment]
@kleinfreund actually no since it is a dup of https://github.com/microsoft/vscode/issues/72042

**Comment by dbaeumer** [comment]
The fix has to happen there. 

As a work around if you start VS Code from a terminal using the `code` command then everything should work as expected.

**Comment by kleinfreund** [comment]
I just wanted to note that the workaround does not work for me.

Running `code` from either zsh or bash still produces `Failed to load the ESLint library for the document …`.

**Comment by dbaeumer** [comment]
That is strange. When you open the developer tools inside VS Code and print `process.env.Path` does this differ from the PATH you see normally in a terminal?

**Comment by kleinfreund** [comment]
No, the `process.env.PATH` property logged via VS Code’s dev tools and my system’s `PATH` environment are identical.

**Comment by dbaeumer** [comment]
This is very wired. We could only reproduce this when the PATH didn't match as outline in microsoft/vscode#72042. I will reopen the issue but I am a little at a loss what causes this on your end.

**Comment by dbaeumer** [comment]
https://github.com/microsoft/vscode/issues/72042 got fixed on the VS Code side. I will close the issue. Please ping if you still see the problem.
